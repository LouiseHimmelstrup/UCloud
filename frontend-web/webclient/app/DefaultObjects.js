import {Cloud} from "../authentication/SDUCloudObject"
import {tusConfig} from "./Configurations";
import Uppy from "uppy";

export const DefaultStatus = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended.",
};

export const RightsMap = {
    NONE: 0,
    READ: 1,
    READ_WRITE: 2,
    OWN: 3,
};

export const AnalysesStatusMap = {
    "PENDING": 0,
    "IN PROGRESS": 1,
    "COMPLETED": 2,
};

export const RightsNameMap = {
    NONE: "None",
    READ: "Read",
    READ_WRITE: "Read/Write",
    OWN: "Own",
};

export const SensitivityLevel = {
    "OPEN_ACCESS": "Open Access",
    "CONFIDENTIAL": "Confidential",
    "SENSITIVE": "Sensitive",
};

export const SensitivityLevelMap = {
    "OPEN_ACCESS": 0,
    "CONFIDENTIAL": 1,
    "SENSITIVE": 2,
};

const initializeUppy = (restrictions) => 
    Uppy.Core({
        autoProceed: false,
        debug: false,
        restrictions: restrictions,
        meta: {
            sensitive: false,
        },
        onBeforeUpload: () => {
            return Cloud.receiveAccessTokenOrRefreshIt().then((data) => {
                tusConfig.headers["Authorization"] = `Bearer ${data}`;
            });
        }
    }).use(Uppy.Tus, tusConfig);

export const initObject = {
    files: {
        files: [],
        filesShown: 10,
        filesPerPage: 10,
        currentFilesPage: 0,
        loading: false,
        path: "",
        filesInfoPath: "",
        filesInfoFile: null,
        projects: []
    },
    uppy: {
        uppyFiles: initializeUppy({maxNumberOfFiles: false}),
        uppyFilesOpen: false,
        uppyRunApp: initializeUppy({ maxNumberOfFiles: 1}),
        uppyRunAppOpen: false
    },
    status: {
        status: DefaultStatus,
        title: ""
    }
};