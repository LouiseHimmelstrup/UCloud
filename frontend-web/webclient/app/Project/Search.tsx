import * as React from "react";
import { Link, Icon } from "ui-components";
import { ProjectMetadata } from "./api";
import * as Heading from "ui-components/Heading";
import { Box, Flex, Text, OldStamp, Divider, Card } from "ui-components";
import { projectViewPage } from "Utilities/ProjectUtilities";

interface SearchItemProps { item: ProjectMetadata }
export const SearchItem = ({ item }: SearchItemProps) => (
    <Card minHeight="168px" p="12px" mb="0.5em" mt="0.5em" borderRadius=".28571429rem">
        <Heading.h3><Link to={projectViewPage(item.sduCloudRoot)}>{item.title}</Link></Heading.h3>

        <Divider />
        <Box mt="1em" mb="1em">
            <Text>
                {firstParagraphWithLimitedLength(
                    defaultIfBlank(item.description, "No description"),
                    800
                )}
            </Text>
        </Box>
        <Divider mb="1em" />
        <OldStamp color="blue" ml="0.2em">
            <Flex>
                <Text color="lightGrey"><Icon name="license" mr="0.5em" size="1.5em"/>{item.license} License</Text>
            </Flex>
        </OldStamp>
    </Card>
);

const defaultIfBlank = (text: string, defaultValue: string): string =>
    text.length == 0 ? defaultValue : text;

const firstParagraphWithLimitedLength = (text: string, maxLength: number): string => {
    const lines = text.split("\n");
    const paragraphEndsAt = lines.findIndex((line) => /^\s*$/.test(line));

    let firstParagraph: string;
    if (paragraphEndsAt == -1) firstParagraph = text;
    else firstParagraph = lines.slice(0, paragraphEndsAt).join("\n");

    if (firstParagraph.length > maxLength) {
        return firstParagraph.substring(0, maxLength) + "...";
    } else {
        return firstParagraph;
    }
};