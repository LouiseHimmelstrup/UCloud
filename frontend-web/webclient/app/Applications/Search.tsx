import * as React from "react";
import {Box, Stamp} from "@/ui-components";
import {emptyPage} from "@/DefaultObjects";
import {joinToString} from "@/UtilityFunctions";
import {useLocation, useNavigate} from "react-router";
import {useEffect} from "react";
import {buildQueryString, getQueryParamOrElse} from "@/Utilities/URIUtilities";
import {useCloudAPI, useCloudCommand} from "@/Authentication/DataHook";
import * as UCloud from "@/UCloud";
import {GridCardGroup} from "@/ui-components/Grid";
import {ApplicationCard} from "@/Applications/Card";
import * as Pagination from "@/Pagination";
import {FilesSearchTabs} from "@/Files/FilesSearchTabs";

interface SearchQuery {
    tags: string[];
    query: string;
    showAllVersions: boolean;
    page: number;
    itemsPerPage: number;
}

function readQuery(queryParams: string): SearchQuery {
    const tags: string[] = [];
    const tagsQuery = getQueryParamOrElse(queryParams, "tags", "");
    tagsQuery.split(",").forEach(it => {
        if (it !== "") {
            tags.push(it);
        }
    });
    const showAllVersions = getQueryParamOrElse(queryParams, "showAllVersions", "false") === "true";
    const query = getQueryParamOrElse(queryParams, "query", "");
    let itemsPerPage = parseInt(getQueryParamOrElse(queryParams, "itemsPerPage", "25"), 10);
    let page = parseInt(getQueryParamOrElse(queryParams, "page", "0"), 10);
    if (isNaN(itemsPerPage) || itemsPerPage <= 0) itemsPerPage = 25;
    if (isNaN(page) || page < 0) page = 0;

    return {query, tags, showAllVersions, itemsPerPage, page};
}

export const SearchResults: React.FunctionComponent<{entriesPerPage: number}> = ({entriesPerPage}) => {
    const navigate = useNavigate();
    const location = useLocation();
    const [, invokeCommand] = useCloudCommand();
    const [results, fetchResults] = useCloudAPI<UCloud.Page<UCloud.compute.ApplicationSummaryWithFavorite>>(
        {noop: true},
        emptyPage
    );

    const queryParams = location.search;
    const parsedQuery = readQuery(queryParams);

    useEffect(() => {
        fetchResults(
            UCloud.compute.apps.searchApps({
                query: new URLSearchParams(queryParams).get("q") ?? "",
                itemsPerPage: 100,
                page: parsedQuery.page,
            })
        );
    }, [queryParams, /* This isn't needed for this one, is it? */entriesPerPage]);

    const toggleFavorite = React.useCallback(async (appName: string, appVersion: string) => {
        await invokeCommand(UCloud.compute.apps.toggleFavorite({appName, appVersion}));
        fetchResults(
            UCloud.compute.apps.searchApps({
                query: new URLSearchParams(queryParams).get("q") ?? "",
                itemsPerPage: 100,
                page: 0
            })
        );
    }, [fetch]);

    return <>
        <FilesSearchTabs active={"APPLICATIONS"} />
        <Pagination.List
            loading={results.loading}
            page={results.data}
            pageRenderer={page => (
                <GridCardGroup>
                    {page.items.map(app => (
                        <ApplicationCard
                            key={`${app.metadata.name}${app.metadata.version}`}
                            app={app}
                            onFavorite={toggleFavorite}
                            isFavorite={app.favorite}
                            tags={app.tags}
                        />))
                    }
                </GridCardGroup>
            )}
            onPageChanged={newPage => {
                navigate(buildQueryString("/applications/search", {
                    q: new URLSearchParams(queryParams).get("q") ?? "",
                    tags: joinToString(parsedQuery.tags),
                    showAllVersions: parsedQuery.showAllVersions.toString(),
                    page: newPage.toString(),
                    itemsPerPage: parsedQuery.itemsPerPage.toString(),
                }));
            }}
        />
    </>;
};
