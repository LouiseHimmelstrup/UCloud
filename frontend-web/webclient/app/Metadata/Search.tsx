import * as React from "react";
import { Link } from "react-router-dom";
import { simpleSearch, ProjectMetadata } from "./api";
import { Page } from "Types";
import { emptyPage } from "DefaultObjects";
import { withRouter } from "react-router-dom";
import * as PropTypes from "prop-types";
import * as Pagination from "Pagination";
import { KeyCode } from "DefaultObjects";
import { History } from "history";
import { updatePageTitle } from "Navigation/Redux/StatusActions";
import * as Heading from "ui-components/Heading";
import { Hide, Input, Box, Flex, Text, Stamp, Divider, Card } from "ui-components";

interface SearchState {
    query: string
    dataProvider: (page: number, itemsPerPage: number) => Promise<Page<ProjectMetadata>>
}

interface SearchProps {
    history: History
    match: {
        params: {
            query?: string
        }
    }
}

class SearchComponent extends React.Component<SearchProps, SearchState> {
    constructor(props, ctx) {
        super(props);
        this.state = {
            query: "",
            dataProvider: (page: number, itemsPerPage: number) => Promise.resolve(emptyPage)
        };
        ctx.store.dispatch(updatePageTitle("Search"));
    }

    static contextTypes = {
        store: PropTypes.object
    }

    componentDidMount() {
        this.checkQueryParams();
    }

    componentDidUpdate() {
        // TODO Doesn't work correctly with react router on back/forwards browser navigation
        this.checkQueryParams();
    }

    private checkQueryParams() {
        const query = this.props.match.params.query;
        if (query == null) {
            return;
        } else if (this.state.query != query) {
            this.setState(() => ({
                query,
                dataProvider: (page: number, itemsPerPage: number) => simpleSearch(query.toLowerCase(), page, itemsPerPage)
            }));
        }
    }

    render() {
        const { history } = this.props;
        return (
            <div>
                <Hide xl lg md>
                    <Input
                        width="100%"
                        placeholder="Search..."
                        onKeyDown={(e: any) => { if (e.keyCode === KeyCode.ENTER) history.push(`/metadata/search/${e.target.value}`) }}
                    />
                </Hide>
                <Heading.h2>Results matching '{this.state.query}'</Heading.h2>

                <Pagination.ManagedList
                    dataProvider={this.state.dataProvider}
                    pageRenderer={this.pageRenderer}
                />
            </div>
        );
    }

    pageRenderer = (page: Page<ProjectMetadata>): React.ReactNode => (
        <>{page.items.map((item, index) => <SearchItem item={item} key={index} />)}</>
    )
}

export const SearchItem = ({ item }: { item: ProjectMetadata }) => (
    <Card height="154px" p="12px" mb="0.5em" mt="0.5em" borderRadius=".28571429rem">
        <Heading.h3><Link to={`/metadata/${item.sduCloudRoot}`}>{item.title}</Link></Heading.h3>

        <Divider />
        <Box mt="1em" mb="1em">
            <Text>
                {firstParagraphWithLimitedLength(
                    defaultIfBlank(item.description, "No description"),
                    800
                )}
            </Text>
        </Box>
        <Divider mb="1em" />
        <Stamp bg="green" color="white" borderColor="green">
            <Box pl="0.5em" pr="0.5em">
                <i className="fas fa-folder-open" />
            </Box>
            Open Access
            </Stamp>
        <Stamp bg="blue" color="white" borderColor="blue" ml="0.2em">
            <Flex>
                <Box pl="0.5em" pr="0.5em">
                    <i className="fas fa-book"></i>
                </Box>
                MIT
                <Text pl="0.2em" color="lightGrey">License</Text>
            </Flex>
        </Stamp>
    </Card>
);

const defaultIfBlank = (text: string, defaultValue: string): string => {
    if (text.length == 0) return defaultValue;
    else return text;
};

const firstParagraphWithLimitedLength = (text: string, maxLength: number): string => {
    const lines = text.split("\n");
    const paragraphEndsAt = lines.findIndex((line) => /^\s*$/.test(line));

    let firstParagraph: string;
    if (paragraphEndsAt == -1) firstParagraph = text;
    else firstParagraph = lines.slice(0, paragraphEndsAt).join("\n");

    if (firstParagraph.length > maxLength) {
        return firstParagraph.substring(0, maxLength) + "...";
    } else {
        return firstParagraph;
    }
};

export const Search = withRouter(SearchComponent);