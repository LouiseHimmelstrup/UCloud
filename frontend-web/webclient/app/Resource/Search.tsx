import * as React from "react";
import {Resource, ResourceApi} from "@/UCloud/ResourceApi";
import {useHistory} from "react-router";
import {useSearch, useSearchPlaceholder} from "@/DefaultObjects";
import {useCallback} from "react";
import {buildQueryString} from "@/Utilities/URIUtilities";

export function useResourceSearch<Res extends Resource>(api: ResourceApi<Res, never>) {
    const history = useHistory();
    const onSearch = useCallback((q) => {
        if (q === "") {
            history.push(`/${api.routingNamespace}`);
        } else {
            history.push(buildQueryString(`/${api.routingNamespace}/search`, {q}));
        }
    }, [api]);

    useSearch(onSearch);
    const searchPlaceholder = `Search ${api.titlePlural.toLowerCase()}...`;
    useSearchPlaceholder(searchPlaceholder);
}
