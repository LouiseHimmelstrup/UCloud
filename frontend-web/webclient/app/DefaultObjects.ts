import {SidebarOption, Page} from "Types";
import {Status} from "Navigation";
import {Analysis, DetailedApplicationSearchReduxState, RunsSortBy} from "Applications";
import {File, DetailedFileSearchReduxState} from "Files";
import {SortOrder, SortBy} from "Files";
import {DashboardStateProps} from "Dashboard";
import {Publication} from "Zenodo";
import {Notification} from "Notifications";
import {Upload} from "Uploader";
import {Activity, ActivityGroup, ActivityFilter} from "Activity";
import {Reducer} from "redux";
import {SimpleSearchStateProps} from "Search";
import * as ApplicationRedux from "Applications/Redux";
import * as AccountingRedux from "Accounting/Redux";
import * as SnackbarRedux from "Snackbar/Redux";
import * as FavoritesRedux from "Favorites/Redux";
import {defaultAvatar} from "UserSettings/Avataaar";
import {SidebarPages} from "ui-components/Sidebar";
import {ScrollResult} from "Scroll/Types";
import * as ProjectRedux from "Project/Redux";

export const DefaultStatus: Status = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended."
};

export enum KeyCode {
    ENTER = 13,
    ESC = 27
}

export const emptyPage: Page<any> = {items: [], itemsPerPage: 25, itemsInTotal: 0, pageNumber: 0, pagesInTotal: 0};

export enum AnalysesStatusMap {
    "PENDING",
    "IN PROGRESS",
    "COMPLETED"
}

export enum RightsNameMap {
    "NONE" = "None",
    "READ" = "Read",
    "READ_WRITE" = "Read/Write"
}

export enum SensitivityLevel {
    "INHERIT" = "Inherit",
    "PRIVATE" = "Private",
    "CONFIDENTIAL" = "Confidential",
    "SENSITIVE" = "Sensitive"
}

export type Sensitivity = keyof typeof SensitivityLevel;

export enum SensitivityLevelMap {
    INHERIT = "INHERIT",
    PRIVATE = "PRIVATE",
    CONFIDENTIAL = "CONFIDENTIAL",
    SENSITIVE = "SENSITIVE"
}

function getFilesSortingColumnOrDefault(columnIndex: 0 | 1): SortBy {
    const sortingColumn = window.localStorage.getItem(`filesSorting${columnIndex}`);
    if (sortingColumn && Object.values(SortBy).includes(sortingColumn)) return sortingColumn as SortBy;
    switch (columnIndex) {
        case 0:
            window.localStorage.setItem("filesSorting0", SortBy.MODIFIED_AT);
            return SortBy.MODIFIED_AT;
        case 1:
            window.localStorage.setItem("filesSorting1", SortBy.SIZE);
            return SortBy.SIZE;
    }
}

function getItemOrDefault<T, T2>(itemName: string, defaultValue: T, en: T2): T {
    const item = window.localStorage.getItem(itemName);
    if (item && Object.values(en).includes(item)) {
        return item as unknown as T;
    }
    return defaultValue;
}

export interface ComponentWithLoadingState {
    loading: boolean
    error?: string
}

export interface ComponentWithPage<T> extends ComponentWithLoadingState {
    page: Page<T>
}

export interface ComponentWithScroll<Item, OffsetType> extends ComponentWithLoadingState {
    scroll?: ScrollResult<Item, OffsetType>
}

export interface ResponsiveReduxObject {
    mediaType: string
    orientation: string
    lessThan: Record<string, boolean>
    greaterThan: Record<string, boolean>
    is: Record<string, boolean>
}

export const initResponsive = (): ResponsiveReduxObject => ({
    mediaType: "",
    orientation: "",
    lessThan: {},
    greaterThan: {},
    is: {},
});

export interface FilePreviewReduxState {
    file?: File
    error?: string
}

export interface FilesReduxObject extends ComponentWithPage<File> {
    sortOrder: SortOrder
    sortBy: SortBy
    path: string
    filesInfoPath: string
    fileSelectorError?: string
    sortingColumns: [SortBy, SortBy]
    fileSelectorLoading: boolean
    fileSelectorShown: boolean
    fileSelectorPage: Page<File>
    fileSelectorPath: string
    fileSelectorCallback: Function
    fileSelectorIsFavorites: boolean
    disallowedPaths: string[]
    invalidPath: boolean
}

export interface FileInfoReduxObject {
    file?: File
    error?: string
    activity: Page<Activity>
    loading: boolean
}

export interface AnalysisReduxObject extends ComponentWithPage<Analysis> {
    sortBy: RunsSortBy
    sortOrder: SortOrder
}

export interface NotificationsReduxObject {
    redirectTo: string
    items: Notification[]
    loading: boolean
    error?: string
}

export interface ZenodoReduxObject extends ComponentWithPage<Publication> {
    connected: boolean
}

export interface StatusReduxObject {
    status: Status
    title: string
    page: SidebarPages
    loading: boolean
}

export interface SidebarReduxObject {
    pp: boolean
    options: SidebarOption[]
    kcCount: number
}

export interface HeaderSearchReduxObject {
    prioritizedSearch: HeaderSearchType
    refresh?: () => void
}

export interface RunApplicationReduxObject {

}

export type ActivityReduxObject = ComponentWithScroll<ActivityGroup, number> & ActivityFilter;

export type HeaderSearchType = "files" | "applications" | "projects";

export interface UploaderReduxObject {
    uploads: Upload[]
    visible: boolean
    path: string
    allowMultiple: boolean
    onFilesUploaded: (p: string) => void
    error?: string
    loading: boolean
}

interface LegacyReducers {
    dashboard?: Reducer<DashboardStateProps>
    files?: Reducer<FilesReduxObject>
    uploader?: Reducer<UploaderReduxObject>
    status?: Reducer<StatusReduxObject>
    notifications?: Reducer<NotificationsReduxObject>
    analyses?: Reducer<AnalysisReduxObject>
    zenodo?: Reducer<ZenodoReduxObject>
    header?: Reducer<HeaderSearchReduxObject>
    sidebar?: Reducer<SidebarReduxObject>
    activity?: Reducer<ActivityReduxObject>
    detailedResult?: Reducer<DetailedResultReduxObject>
}

export type Reducers = LegacyReducers & ApplicationRedux.Reducers & AccountingRedux.Reducers;

export type DetailedResultReduxObject = ComponentWithPage<File>

export const initDetailedResult = (): DetailedResultReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined
});

/* FIXME */
interface LegacyReduxObject {
    dashboard: DashboardStateProps
    files: FilesReduxObject,
    uploader: UploaderReduxObject
    status: StatusReduxObject,
    notifications: NotificationsReduxObject
    analyses: AnalysisReduxObject
    zenodo: ZenodoReduxObject
    header: HeaderSearchReduxObject
    sidebar: SidebarReduxObject
    activity: ActivityReduxObject
    detailedResult: DetailedResultReduxObject
    simpleSearch: SimpleSearchStateProps
    detailedFileSearch: DetailedFileSearchReduxState
    detailedApplicationSearch: DetailedApplicationSearchReduxState
    fileInfo: FileInfoReduxObject
    avatar: AvatarReduxObject
    filePreview: FilePreviewReduxState
    responsive?: ResponsiveReduxObject
    project: ProjectRedux.State
    loading?: boolean
}

export type ReduxObject =
    LegacyReduxObject &
    ApplicationRedux.Objects &
    AccountingRedux.Objects &
    FavoritesRedux.Objects &
    SnackbarRedux.Wrapper;

export const initActivity = (): ActivityReduxObject => ({
    loading: false
});

export const initNotifications = (): NotificationsReduxObject => ({
    items: [],
    loading: false,
    redirectTo: "",
    error: undefined
});

export const initHeader = (): HeaderSearchReduxObject => ({
    prioritizedSearch: "files"
});

export const initStatus = (): StatusReduxObject => ({
    status: DefaultStatus,
    title: "",
    page: SidebarPages.None,
    loading: false
});

export const initDashboard = (): DashboardStateProps => ({
    favoriteFiles: [],
    recentFiles: [],
    recentAnalyses: [],
    notifications: [],
    favoriteLoading: false,
    recentLoading: false,
    analysesLoading: false
});

export function initObject(homeFolder: string): ReduxObject {
    return {
        dashboard: initDashboard(),
        files: initFiles(homeFolder),
        status: initStatus(),
        header: initHeader(),
        notifications: initNotifications(),
        analyses: initAnalyses(),
        zenodo: initZenodo(),
        sidebar: initSidebar(),
        uploader: initUploads(),
        activity: initActivity(),
        detailedResult: initDetailedResult(),
        simpleSearch: initSimpleSearch(),
        detailedApplicationSearch: initApplicationsAdvancedSearch(),
        detailedFileSearch: initFilesDetailedSearch(),
        fileInfo: initFileInfo(),
        avatar: initAvatar(),
        filePreview: initFilePreview(),
        project: ProjectRedux.initialState,
        ...ApplicationRedux.init(),
        ...AccountingRedux.init(),
        ...FavoritesRedux.init(),
        ...SnackbarRedux.init(),
        responsive: undefined,
    }
}


export const initFilePreview = () => ({
    file: undefined,
    error: undefined
});

export type AvatarReduxObject = typeof defaultAvatar & { error?: string };
export const initAvatar = (): AvatarReduxObject => ({...defaultAvatar, error: undefined});

export const initSimpleSearch = (): SimpleSearchStateProps => ({
    files: emptyPage,
    filesLoading: false,
    applications: emptyPage,
    applicationsLoading: false,
    errors: [],
    search: "",
    applicationSearch: initApplicationsAdvancedSearch(),
    fileSearch: initFilesDetailedSearch()
});

export const initAnalyses = (): AnalysisReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined,
    sortBy: RunsSortBy.createdAt,
    sortOrder: SortOrder.ASCENDING
});

export const initZenodo = (): ZenodoReduxObject => ({
    connected: false,
    loading: false,
    page: emptyPage
});

export const initSidebar = (): SidebarReduxObject => ({
    pp: false,
    kcCount: 0,
    options: []
});

export const initUploads = (): UploaderReduxObject => ({
    path: "",
    uploads: [],
    visible: false,
    allowMultiple: false,
    error: undefined,
    onFilesUploaded: () => null,
    loading: false
});

export const initFileInfo = (): FileInfoReduxObject => ({
    activity: emptyPage,
    loading: false
});

export const initFiles = (homeFolder: string): FilesReduxObject => ({
    page: emptyPage,
    sortOrder: getItemOrDefault("sortOrder", SortOrder.ASCENDING, SortOrder),
    sortBy: getItemOrDefault("sortBy", SortBy.PATH, SortBy),
    loading: false,
    error: undefined,
    path: "",
    filesInfoPath: "",
    sortingColumns: [getFilesSortingColumnOrDefault(0), getFilesSortingColumnOrDefault(1)],
    fileSelectorLoading: false,
    fileSelectorShown: false,
    fileSelectorPage: emptyPage,
    fileSelectorPath: homeFolder,
    fileSelectorIsFavorites: false,
    fileSelectorCallback: () => undefined,
    fileSelectorError: undefined,
    disallowedPaths: [],
    invalidPath: false
});

export const initFilesDetailedSearch = (): DetailedFileSearchReduxState => ({
    hidden: true,
    allowFolders: true,
    allowFiles: true,
    fileName: "",
    extensions: new Set(),
    tags: new Set(),
    sensitivities: new Set(),
    createdBefore: undefined,
    createdAfter: undefined,
    modifiedBefore: undefined,
    modifiedAfter: undefined,
    error: undefined,
    loading: false
});

export const initApplicationsAdvancedSearch = (): DetailedApplicationSearchReduxState => ({
    error: undefined,
    loading: false,
    hidden: true,
    appName: "",
    appVersion: "",
    tags: ""
});
