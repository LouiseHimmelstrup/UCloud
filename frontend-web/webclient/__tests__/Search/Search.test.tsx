import * as React from "react";
import Search from "../../app/Search/Search";
import { create } from "react-test-renderer";
import { createMemoryHistory } from "history";
import { MemoryRouter } from "react-router";
import { configureStore, responsive } from "../../app/Utilities/ReduxUtilities";
import files from "../../app/Files/Redux/FilesReducer";
import { initSimpleSearch, initFilesDetailedSearch, initApplicationsAdvancedSearch } from "../../app/DefaultObjects";
import theme from "../../app/ui-components/theme";
import { Provider } from "react-redux";
import simpleSearch from "../../app/Search/Redux/SearchReducer";
import detailedFileSearch from "../../app/Files/Redux/DetailedFileSearchReducer";
import detailedApplicationSearch from "../../app/Applications/Redux/DetailedApplicationSearchReducer";
import { ThemeProvider } from "styled-components";
import "jest-styled-components";


describe("Search", () => {
    test.skip("Mount", () => {
        expect(create(
            <Provider store={configureStore({
                simpleSearch: initSimpleSearch(),
                detailedFileSearch: initFilesDetailedSearch(),
                detailedApplicationSearch: initApplicationsAdvancedSearch()
            }, {
                    simpleSearch,
                    detailedFileSearch,
                    detailedApplicationSearch,
                    files,
                    responsive
                })}>
                <ThemeProvider theme={theme}>
                    <MemoryRouter>
                        <Search
                            match={{ params: { priority: "" }, isExact: false, path: "", url: "" }}
                            history={createMemoryHistory()}
                            location={{ search: "" }}
                        />
                    </MemoryRouter>
                </ThemeProvider>
            </Provider>)).toMatchSnapshot();
    });
})