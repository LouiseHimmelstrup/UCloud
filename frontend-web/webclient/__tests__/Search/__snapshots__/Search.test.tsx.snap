// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Mount 1`] = `
<div
  className="sc-bdVaJa fENTec"
  cursor="auto"
>
  <div
    className="sc-bdVaJa sc-bwzfXH sc-eHgmQL ZdyQV"
    cursor="auto"
    height={96}
    width={1}
  >
    <div
      className="sc-EHOje sc-feJyhm iqoSv"
    >
      <div
        className="sc-bZQynM sc-iELTvK DrRrL"
        cursor="pointer"
        fontSize={2}
        onClick={[Function]}
        selected={false}
      >
        Files
      </div>
      <div
        className="sc-bZQynM sc-iELTvK DrRrL"
        cursor="pointer"
        fontSize={2}
        onClick={[Function]}
        selected={false}
      >
        Applications
      </div>
    </div>
    <div
      className="sc-EHOje fzjOHr"
      cursor="inherit"
    >
      <div
        className="sc-bdVaJa ehOkeT"
        cursor="auto"
      />
      <div
        className="sc-tilXH hnDjpX"
        data-tag="dropdown"
      >
        <span
          className="sc-bZQynM lphChX"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="sc-bZQynM fufZuQ"
            cursor="inherit"
          >
             
             per page 25
          </span>
          <svg
            className="sc-htoDjs fZlZAn"
            clipRule="evenodd"
            cursor="inherit"
            fill="currentcolor"
            fillRule="evenodd"
            height=".7em"
            ml=".7em"
            viewBox="0 0 24 12"
            width=".7em"
          >
            <path
              d="M0 0l11.959 12 12-12H0z"
            />
          </svg>
        </span>
        <div
          className="sc-hEsumM dBaQvi"
          color="black"
          cursor="pointer"
          disabled={false}
          onClick={[Function]}
          overflow="visible"
          width="80px"
        >
          <div
            className="sc-bZQynM hdeUCE"
            cursor="pointer"
            onClick={[Function]}
          >
            10
          </div>
          <div
            className="sc-bZQynM hdeUCE"
            cursor="pointer"
            onClick={[Function]}
          >
            25
          </div>
          <div
            className="sc-bZQynM hdeUCE"
            cursor="pointer"
            onClick={[Function]}
          >
            50
          </div>
          <div
            className="sc-bZQynM hdeUCE"
            cursor="pointer"
            onClick={[Function]}
          >
            100
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa ikztbv"
    cursor="auto"
  >
    <div
      className="sc-bdVaJa sc-jhAzac eTvuVp"
    />
  </div>
</div>
`;
